tests/virt_autotest/libvirt_extend_storage_lvm.pm

```diff
@@ -0,0 +1,165 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Virtualization Extend LVM Storage pool / volume test
+#
+# Test flow:
+# - Check a additional free hard disk
+# - Wipe hard disk clean
+# - Partition test disk
+# - Create a pv and display result
+# - Create a vg named 'lvm_vg' and display result
+# - Create a storage pool (lv) named 'guest_image_lvm'
+# - Create a logical volume (lv) size 1G
+# - Attach a logical volume (lv) to guest systems
+# - Clone a logical volume (lv)
+# - Cleanup
+# Maintainer: Leon Guo <xguo@suse.com>
+
+use base "virt_feature_test_base";
+use virt_autotest::utils;
+use virt_autotest::common;
+use strict;
+use warnings;
+use testapi;
+use utils;
+use virt_utils;
+use version_utils 'is_sle';
+
+our $lvm_vg_name = 'lvm_vg';
+our $lvm_pool_name = 'guest_image_lvm';
+our $timeout = 180;
+sub run_test {
+    my ($self) = @_;
+
+    record_info "Prepare Guest Systems";
+    foreach (keys %virt_autotest::common::guests) {
+        start_guests() unless is_guest_online($_);
+    }
+
+    ## About LVM volumes management
+    my ($sys_disk, $virt_disk, $dev, $lvm_disk);
+    # Check with all existed Hard disks
+    my @disks = split(/\n/, script_output("lsblk -n -l -o NAME -d -e 7,11"));
+    my $scalar = @disks;
+    # Requires at least 2 physical hard disks for LVM Storage test
+    if (($scalar eq 1) || get_var('KEEP_DISKS')) {
+        record_info("WARNING", "Requires at least 2 physical hard disks for LVM Storage test\n", result => 'softfail');
+        return;
+    }
+    # Use a additional free hard disk for LVM volumes
+    $sys_disk = script_output 'lsblk -rnoPKNAME $(findmnt -nrvoSOURCE /)';
+    $virt_disk = script_output 'lsblk -rnoPKNAME $(findmnt -nrvoSOURCE /var/lib/libvirt)';
+    record_info "System Disk:", "$sys_disk";
+    record_info "Virt Disk:", "$virt_disk";
+    record_info("List Block Devices", script_output("lsblk"));
+    $dev = "/dev/";
+    foreach my $disk (@disks) {
+        if ($disk ne $sys_disk) {
+            if ($disk ne $virt_disk) {
```

Comment:
These two if can be combined into one
```
 if ($disk ne $sys_disk) and ($disk ne $virt_disk)
```

Reply:
yes, but, currently osd tidy-check occur ERROR with the above code change. currently, no idea.

Reply:
figure out the solution to fix up osd tidy-check occur. update code to find out unused hard disk now.